src/coinjoin/context.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023 The Dash Core developers
+// Distributed under the MIT/X11 software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_COINJOIN_CONTEXT_H
+#define BITCOIN_COINJOIN_CONTEXT_H
+
+#if defined(HAVE_CONFIG_H)
+#include <config/bitcoin-config.h>
+#endif
+
+#include <memory>
+
+class CBlockPolicyEstimator;
+#ifdef ENABLE_WALLET
+class CCoinJoinClientQueueManager;
+#endif // ENABLE_WALLET
+class CCoinJoinServer;
+class CConnman;
+class CMasternodeSync;
+class CTxMemPool;
+
+#ifdef ENABLE_WALLET
+struct CJClientManager;
+#endif // ENABLE_WALLET
+
+struct CJContext {
+    CJContext() = delete;
```

Comment:
why you can't copy CJContext if all data is shared_ptr? Shared_ptr means that you can copy objects of this class that many times as you need. If copying is forbidden -> better to use unique_ptr or something similar. 

Can you provide clarification for how many CJContext can exist independently on same time also?

If more than one global variable  ::coinJoinServer coule be re-written everytime when new object is created.

Reply:
You can only have one copy. I don't recollect `CJContext` as ever being `shared_ptr`, its members were `shared_ptr` as they were being shared between the global and the object member but they pointed to the same object (and when deglobalized, they were moved back to being `unique_ptr`). The program does not expect more than one instance.

Though, restructuring `CJContext` to be more `LLMQContext` like may render the concern moot, since all are `unique_ptr` now.